dispatch/dispatch.go

```diff
@@ -287,6 +298,7 @@ func (d *Dispatcher) unsubscribe(id uuid.UUID, usedChan <-chan interface{}) erro
 		pipes[i] = pipes[len(pipes)-1]
 		pipes[len(pipes)-1] = nil
 		d.routes[id] = pipes[:len(pipes)-1]
+		atomic.AddInt32(&d.subscriberCount, -1)
```

Comment:
Suggestion: maybe a check to only decrement if its above 0? If you think that's unnecessary all good just resolve.

Reply:
I like the thought to ensure smooth running, but if it manages to go negative, it'll be a nice indicator to highlight that something else is amiss